### PR TITLE
Add unit-tests for atan() functions (math.h) 

### LINKED ISSUE #3268: Add unit-tests for atan() functions (math.h)
Look at https://github.com/embox/embox/blob/master/src/compat/libc/math/tests/sqrt_tests.c

### PR DESCRIPTION
**Pull request by jiuyueshenhua**
This test module is also added to the x86 template
closes #3268

### CODE REVIEW

**Review by anton-bondarev — APPROVED** [review]
LGTM

File: src/compat/libc/math/tests/atan_tests.c [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+ * @date May 20, 2024
+ * @author liu kang bing
+ */
+
+#include <math.h>
+
+#include <embox/test.h>
+EMBOX_TEST_SUITE("atan() tests");
+
+bool is_equal(double x, double y) {
```
Comment (anton-bondarev) [inline src/compat/libc/math/tests/atan_tests.c:13]: Could you make it 'static'

**Comment by anton-bondarev** [comment]
@jiuyueshenhua Thanks
I've approved it.
However I wrote a small comment. Could you fix it?
And then please prepare the PR to merge (rebase under upstream/master)

File: src/compat/libc/math/tests/atan_tests.c (hunk 1 above)
Comment (jiuyueshenhua) [reply src/compat/libc/math/tests/atan_tests.c:13]: > Could you make it 'static'

you mean `static bool is_equal(d,d)`?

**Comment by jiuyueshenhua** [comment]
> @jiuyueshenhua Thanks I've approved it. However I wrote a small comment. Could you fix it? And then please prepare the PR to merge (rebase under upstream/master)

I have completed it according to your requests

**Comment by anton-bondarev** [comment]
> I have completed it according to your requests

@jiuyueshenhua Thanks a lot
Could you also squash [libc/math:(test) Add atan test](https://github.com/embox/embox/pull/3275/commits/27df12c02cca543b0679371d25ed2eb32fd0f4d7) and [src/compat.../tests: (fix) add static to helper fuction](https://github.com/embox/embox/pull/3275/commits/907d4cd130484d916c83e108ec63ef1a5ab3313a)

**Comment by anton-bondarev** [comment]
> Is that ok?

Yes. It's OK
Thanks a lot.
I've merged it

**Comment by anton-bondarev** [comment]
> Thank you for your patience.

No problem! :)

**Pull request merged**